### PR TITLE
Python API for `place_pwlin`.

### PR DESCRIPTION
**Pull request by halfflat**
* Add Python bindings for the `place_pwlin` and `isometry` classes.
* Add equality test for Python `mpoint`.
* Add unit tests for new Python interfaces.
* Split C++ API morphology documentation into its own file.
* Add C++ API and Python API documentation for `place_pwlin` and `isometry`.

### CODE REVIEW

**Review by noraabiakar — CHANGES_REQUESTED** [review]
Looks good to me! just one minor comment in the docs.

File: doc/python/morphology.rst [hunk 1]
```diff
@@ -391,4 +391,92 @@ Cell morphology
...
+    .. py:method:: __call__(point: mpoint) -> mpoint
+
+       Apply the isometry to given point.
+
+    .. py:method:: __call__(point: Tuple[float, float, float, ...]) -> Tuple[float, float, float, ...]
+       :noindex:
+
+       Apply the isometry to the first three components of the given tuple, interpreted as a point.
+
+    .. py:function:: __mul__(a: isometry, b: isometry) -> isometry
```
Comment (bcumming) [inline doc/python/morphology.rst:478]: Is `->` how one specifies the return type?
I have been using `:rtype`, e.g.:
https://github.com/arbor-sim/arbor/blame/master/doc/python/morphology.rst#L368

File: python/morphology.cpp [hunk 2]
```diff
@@ -60,11 +66,8 @@ void register_morphology(py::module& m) {
     // arb::mpoint
     py::class_<arb::mpoint> mpoint(m, "mpoint");
     mpoint
-        .def(py::init(
-                [](double x, double y, double z, double r) {
-                    return arb::mpoint{x,y,z,r};
-                }),
-                "x"_a, "y"_a, "z"_a, "radius"_a, "All values in μm.")
+        .def(py::init<double, double, double, double>(),
+             "x"_a, "y"_a, "z"_a, "radius"_a)
```
Comment (bcumming) [inline python/morphology.cpp:70]: The docstring has been removed (which stated that all inputs were in um)

File: python/morphology.cpp [hunk 3]
```diff
@@ -112,6 +116,68 @@ void register_morphology(py::module& m) {
         .def("__str__", [](const arb::mcable& c) { return util::pprintf("{}", c); })
         .def("__repr__", [](const arb::mcable& c) { return util::pprintf("{}", c); });
 
+    // arb::isometry
+    py::class_<arb::isometry> isometry(m, "isometry");
+    isometry
```
Comment (bcumming) [inline python/morphology.cpp:121]: docstrings are missing for the calls and methods in the `isometry` wrapper.

File: python/morphology.cpp [hunk 4]
```diff
@@ -112,6 +116,68 @@ void register_morphology(py::module& m) {
...
+            "theta"_a, "x"_a, "y"_a, "z"_a)
+        .def_static("rotate",
+            [](double theta, py::tuple t) {
+                if (py::len(t)!=3) throw std::runtime_error("tuple length != 3");
+                return arb::isometry::rotate(theta, t[0].cast<double>(), t[1].cast<double>(), t[2].cast<double>());
+            },
+            "theta"_a, "axis"_a);
+
+    // arb::place_pwlin
+    py::class_<arb::place_pwlin> place(m, "place_pwlin");
```
Comment (bcumming) [inline python/morphology.cpp:164]: Missing docstrings also for methods of `place_pwlin`.

**Review by bcumming — CHANGES_REQUESTED** [review]
Looks good.
The Python wrappers are missing docstrings.

File: doc/python/morphology.rst (hunk 1 above)
Comment (halfflat) [reply doc/python/morphology.rst:478]: I just got that syntax from the Python type annotation syntax. I don't know if it's the right or even better way to do it though.

**Review by bcumming — APPROVED** [review]

**Review by noraabiakar — APPROVED** [review]

**Pull request merged**